mne/viz/ica.py

```diff
@@ -1146,26 +1149,58 @@ def _plot_ica_overlay_raw(data, data_cln, times, title, ch_types_used, show):
     """
     import matplotlib.pyplot as plt
 
-    # Restore sensor space data and keep all PCA components
-    # let's now compare the date before and after cleaning.
-    # first the raw data
-    assert data.shape == data_cln.shape
-    fig, (ax1, ax2) = plt.subplots(2, 1, sharex=True)
-    plt.suptitle(title)
-    ax1.plot(times, data.T, color="r")
-    ax1.plot(times, data_cln.T, color="k")
-    ax1.set(xlabel="Time (s)", xlim=times[[0, -1]], title="Raw data")
-
-    _ch_types = {"mag": "Magnetometers", "grad": "Gradiometers", "eeg": "EEG"}
-    ch_types = ", ".join([_ch_types[k] for k in ch_types_used])
-    ax2.set_title("Average across channels ({})".format(ch_types))
-    ax2.plot(times, data.mean(0), color="r")
-    ax2.plot(times, data_cln.mean(0), color="k")
-    ax2.set(xlabel="Time (s)", xlim=times[[0, -1]])
-    tight_layout(fig=fig)
+    ch_types = raw.get_channel_types(picks=picks, unique=True)
+    for ch_type in ch_types:
+        if ch_type in ("mag", "grad"):
+            fig, ax = plt.subplots(3, 1, sharex=True)
+        elif ch_type == "eeg" and not _has_eeg_average_ref_proj(raw.info):
+            fig, ax = plt.subplots(3, 1, sharex=True)
+        else:
+            fig, ax = plt.subplots(2, 1, sharex=True)
+        plt.suptitle(title)
+
+        # select sensors and retrieve data array
+        picks_by_type = _picks_to_idx(raw.info, ch_type, exclude=())
+        picks_ = np.intersect1d(picks, picks_by_type)
+        data, times = raw[picks_, start:stop]
+        data_cln, _ = raw_cln[picks_, start:stop]
+
+        # plot all sensors of the same type together
+        ax[0].plot(times, data.T, color="r")
+        ax[0].plot(times, data_cln.T, color="k")
+        _ch_type = (
+            DEFAULTS["titles"][ch_type] if ch_type in DEFAULTS["titles"] else ch_type
+        )
+        ax[0].set(xlabel="Time (s)", xlim=times[[0, -1]], title=f"Raw {_ch_type} data")
```

Comment:
This feels a bit more readable to me, though I guess it might be slightly less accurate? (e.g., `times[0]` might not exactly equal `start` depending on sample timing).  On the other hand, even if less accurate, the plot will better reflect what the user asked for (i.e. if they want `start=1.5` the x-axis first tick should say `1.5` not `1.50013` or whatever)

```suggestion
        ax[0].set(xlabel="Time (s)", xlim=(start, stop), title=f"Raw {_ch_type} data")
```

Reply:
Actually, start/stop are defined in samples at that point. Thus, I see 3 options:
- forward the requested start/stop in seconds to that function
- convert back start/stop to seconds (should yield the same value as `times[[0, -1]]`)
- leave it as is

I'm +1 to leave it as is for the little difference it would make depending on the sampling rate, e.g. `1.5` vs `1.50013`.

Reply:
ah,ok,leave as-is then